downstream-keyer.cpp

```diff
@@ -914,10 +915,13 @@ bool DownstreamKeyer::SwitchToScene(QString scene_name)
 	return false;
 }
 
-void DownstreamKeyer::add_scene(QString scene_name, obs_source_t *s)
+void DownstreamKeyer::add_scene(QString scene_name, obs_source_t *s, int insertBeforeRow)
 {
 	const auto item = new QListWidgetItem(scene_name);
-	scenesList->addItem(item);
+	if (insertBeforeRow < 0) {
```

Comment:
Also check if `insertBeforeRow > scenesList->count()` ?

Reply:
Good call. Thank you!